systemtests/scripts/setup

```diff
@@ -13,6 +13,9 @@ print_debug "creating database \"${db_name}\" (${DBTYPE})"
 "${scripts}/make_bareos_tables" "${DBTYPE}" >/dev/null 2>&1
 "${scripts}/grant_bareos_privileges" "${DBTYPE}" >/dev/null 2>&1
 
+# Fill ${BackupDirectory} with data.
+setup_data
+
```

Comment:
this will run setup_data for every systemtest that calls `${rscripts}/setup` is this intended?

Reply:
In the old way (testrunner without subtests), the testrunner would setup the environment as well as the database, setup the custom data, and then finally run the bareos binaries.

When we do the subtest, the routine changes: the `test-setup` script sets up the environment as well as the database, but starts running the binaries without setting up the custom data, which falls in the hands of the new `testrunner-<your-test-name>`. At first hand, this seemed not important, as other systemtests that need data have no problem with it. But with the `scheduler-backup` it is different. When running the test, it complains about not finding the custom data. As far as I've understood, this happens because the first job of the scheduler runs as soon as the director starts. That happens quickly before the testrunner deploys the custom data. So, when trying the find the data to backup, bareos stalls, as there seems to be an issue with the sd and fd, and things get stuck there, which makes the `wait` command stuck too, which in turn gets the whole test stuck.

Setting up the custom data for the test *before* running the binaries, makes sure the first scheduled job runs properly on the start of director, and the rest of the test goes smoothly.

Other tricks like disabling and enabling the scheduler, had no impact, and then doing this change made the most sense for me.

It felt setting up the data in the setup script would make sense. Even if other tests don't need data, it doesn't cost too much in setup time, plus it could be useful for future ramifications of the test or extra subtests added.

Another solution would have been to leave the `scheduler-backup` systemtest as is, and create a whole new test, but I think for the functionality we're adding, it feels too much overkill, and converting the whole thing into subtests felt better.

let me know what you think of this

Reply:
Another useful benefit of having `setup_data` in the `setup` script I just encountered, is that within any systemtest, you could run `test-setup` and have an environment ready to use, without having to run the entire testrunner. This gives more flexibility when experimenting within the environment of a certain test, or for debugging or anything of that sort

Reply:
After a conversation with Joerg, I could handle this issue without having to setup the data in the `${rscripts}/setup` script.